prov/util/src/util_shm.c

```diff
@@ -402,6 +420,9 @@ void smr_map_del(struct smr_map *map, int id)
 	if (!entry)
 		munmap(map->peers[id].region, map->peers[id].region->total_size);
 
+	(void) ofi_rbmap_find_delete(&map->rbmap,
+				     (void *) map->peers[id].peer.name);
```

Comment:
Do you need map->lock here?

Reply:
smr_map_del and smr_map_add assume that the map->lock is acquired already. There's one place it's not so I'll fix that an add a comment to say it should be held.